app/src/main/java/com/humberto/tasky/agenda/data/task/TaskRepositoryImpl.kt

```diff
@@ -23,14 +29,37 @@ class TaskRepositoryImpl @Inject constructor(
 
     override suspend fun createTask(task: AgendaItem.Task): EmptyResult<DataError> {
         return try {
-            taskDao.upsertTask(task.toTaskEntity())
-            Result.Success(Unit)
+            val result = safeCall {
+                taskDao.upsertTask(task.toTaskEntity())
+                agendaApi.createTask(task.toTaskRequest())
+            }.onError { error ->
+                if(error == DataError.Network.NO_INTERNET) {
+                    taskDao.insertModifiedTask(
+                        task.toModifiedTaskEntity(
+                            modificationType = ModificationType.Created
+                        )
+                    )
+                    return Result.Success(Unit)
+                }
+            }
+            return result
         } catch (e: SQLiteFullException) {
             Result.Error(DataError.Local.DISK_FULL)
         }
     }
 
     override suspend fun deleteTask(taskId: String) {
-        taskDao.deleteTask(taskId)
+        val result = safeCall {
+            taskDao.deleteTask(taskId)
+            agendaApi.deleteTask(taskId)
```

Comment:
Same here; only attempt the API call when the DB call succeeds. Being explicit about it helps maintain a codebase and avoid potentially hard-to-find bugs. This current approach relies on `apiCall` to catch an exception from `taskDao.deleteTask()` (what if a new Room lib release no longer throws an exception when an entry can’t be made?) not to call the API.